modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ReadOnlyUnbackedObservableList.java

```diff
@@ -113,7 +113,7 @@ public void callObservers(Change<E> c) {
     @Override public int indexOf(Object o) {
         if (o == null) return -1;
 
-        for (int i = 0; i < size(); i++) {
+        for (int i = 0, max = size(); i < max; i++) {
             Object obj = get(i);
             if (o.equals(obj)) return i;
         }
```

Comment:
An alternative is
```
        return IntStream.range(0, size())
            .filter(i -> o.equals(get(i)))
            .findFirst()
            .orElse(-1);
```
I don't know if it helps.

Reply:
the for-loop is certainly faster and would allocate less memory - i find the `for(int i = 0, max = size())`-style a bit odd

Reply:
I could move `int max = size();` outside the loop

Reply:
But why? The initialization block of a `for` statement is exactly where you'd put loop-scoped variables.

Reply:
This change significantly improves performance because the BitSet's size () method is an O (N) implementation. You may be wondering because it is O (1) in many other collections.